charts/daemonset-app/templates/_helpers.tpl

```diff
@@ -0,0 +1,16 @@
+{{/*
+This function generates an extended set of labels by combining the base labels 
+from the "nd-common.labels" template with additional custom labels. 
+
+The additional labels include:
+  - helm.chart/name: Specifies the name of the chart (hardcoded as "daemonset-app").
+  - helm.chart/version: Includes the chart version dynamically from .Chart.Version.
+*/}}
+{{- define "daemonset-app.labels" -}}
+{{- $baseLabels := include "nd-common.labels" . | fromYaml -}}
+{{- $extendedLabels := merge $baseLabels (dict
+    "helm.sh/chartName" "daemonset-app"
+    "helm.sh/chartVersion" .Chart.Version
```

Comment:
Adding standup discussion comments for the record...

I suggested we do not use the helm.sh namespace and instead use a namespace like helm.nextdoor.com or oss.nextdoor.com. This is because we are adding a _new_ label that helm.sh themselves have not yet defined and therefore no API for it exists. In order to avoid any future naming collisions, I believe we should be name-spacing this new label use-case under our own namespace.

This might be an over-optimization, so we decided to differ to status quo due to no strong opinions.

cc: @diranged

Reply:
Follow up PR https://github.com/Nextdoor/k8s-charts/pull/351